api/helpers/gamestate/moves/resolve/two.js

```diff
@@ -0,0 +1,26 @@
+module.exports = {
+  friendlyName: 'Resolve Two One-Off',
+
+  description: 'Returns new GameState resulting from resolving a two, which scraps target royal and glasses eight on the field, and counter one-off cards.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'The latest GameState before the two resolves',
+      required: true,
+    },
+  },
+  sync: true, // synchronous helper
+  fn: ({ currentState }, exits) => {
+    let result = _.cloneDeep(currentState);
+
+    // remove target card from opponent hand 
+    const targetPlayedIndex = result.p1.faceCards.findIndex(({ id }) => id === result.targetCard.id);
+    result.p1.faceCards.splice(targetPlayedIndex, 1);
```

Comment:
It is also possible to target a jack with a 2, and that case is a bit more complicated. We should use `res.oneOffTargetType`  (which will either be 'faceCard' or 'jack') to identify whether we are targeting a jack. If it's a faceCard, do exactly what you have here.

If it's a jack, we need to loop through the opponent's points, and for each point card, check its attachments to look for the targetCard. Then we remove the jack from its point card's attachments, put the jack in the scrap, and move the point card from the opponent's points to the player's points

Reply:
in order to test this i will have to wait for this #1045  to mege so that jack can be playeable as of now it gives movetype off jack is invalid.